ACPs/77-reinventing-subnets/README.md

```diff
@@ -213,167 +281,65 @@ An EVM Subnet may choose to implement this step like so:
+If all Subnet Validators are removed, there are no valid Warp messages that can be produced to register new Subnet Validators through `RegisterSubnetValidatorMessage`. With no validators, block production will halt and the Subnet is unrecoverable. To serve as a guardrail against this situation, a Subnet Validator removal will be considered invalid if the validator being removed is the last one. A future ACP can remove this guardrail as users get more familiar with the new Subnet mechanics and tooling matures to fork a Subnet.
 
-The expected path for full removal from a Subnet's validator set is via a `SetSubnetValidatorWeightTx` with weight `0` which requires a Warp message produced by the Subnet. However, the ability to stop participating in Subnet validation is critical for censorship-resistance and/or failed Subnets. If a Subnet Validator wishes to stop participating in its Subnet's consensus, they can do so through this transaction. This is enabled on the P-Chain to prevent Subnets from locking Subnet Validators into participating in consensus indefinitely.
+The `nonce` contained within the `SubnetValidatorWeightMessage` in the transaction must satisfy `nonce >= minNonce`. Note that `nonce` is not required to be incremented by `1` with each successive validator weight update. If `minNonce` is `MaxUint64`, the `weight` in the `SetSubnetValidatorWeightTx` is required to be `0` to prevent Subnets from being unable to remove `nodeID` in a subsequent `SetSubnetValidatorWeightTx`. When a Subnet Validator is removed from the active validator set (`weight == 0`), the `minNonce` and `validationID` will be removed from the P-Chain state. This state can be reaped during validator removal since `validationID` can never be re-initialized as a result of the replay protection provided by `expiry` in `RegisterSubnetValidatorTx`. `minNonce` will be set to `nonce + 1` when `SetSubnetValidatorWeightTx` is executed and `weight != 0`.
 
-Note that this does not modify a Subnet's total staking weight. This transaction only marks the Subnet Validator as inactive but does not remove it from the Subnet's validator set. Inactive Subnet Validators can re-activate at any time by increasing their balance with an `IncreaseBalanceTx`.
+#### `DisableSubnetValidatorTx`
 
-Subnet creators should be aware that there is no notion of `MinStakeDuration` that is enforced by the P-Chain. It is expected that Subnets who choose to enforce a `MinStakeDuration` will lock the validator's Stake for the Subnet's desired `MinStakeDuration`.
+Subnet Validators can use `DisableSubnetValidatorTx` to mark their validator as inactive. The specification for this transaction is:
 
 ```golang
-type DisableValidatorTx struct {
+type DisableSubnetValidatorTx struct {
     // Metadata, inputs and outputs
     BaseTx
     // ID corresponding to the validator
     ValidationID ids.ID `json:"validationID"`
-    // Ed25519 Signature on [TxID]
-    Signature []byte `json:"signature"`
```

Comment:
Now that this can be a multi-sig, we need to add a `DisableAuth verify.Verifiable` here.

Reply:
Thank for calling that out. Added now